firefox-ios/Client/Frontend/Browser/Toolbars/AddressBarPanGestureHandler.swift

```diff
@@ -174,4 +176,19 @@ final class AddressBarPanGestureHandler: NSObject {
         let xTranslation = isSwipingLeft ? width + translation.x + UX.offset : -width + translation.x - UX.offset
         webPagePreview.transform = CGAffineTransform(translationX: xTranslation, y: 0)
     }
+
+    /// Calculates the index of the next tab to display based on the current index, swipe direction, and layout direction.
+    /// This function ensures that tab navigation behaves intuitively for
+    /// both left-to-right (LTR) and right-to-left (RTL) user interfaces.
+    /// Swiping left advances to the next tab in LTR, but to the previous tab in RTL, and vice versa.
+    private func nextTabIndex(from index: Int, isSwipingLeft: Bool) -> Int {
+        let isRTL = UIView.userInterfaceLayoutDirection(
```

Comment:
Can we please add unit tests for this?

Reply:
There is already a unit test ticket for the entire `AddressBarPanGestureHandler` class which I'll address further.